stytra/experiments/tracking_experiments.py

```diff
@@ -84,49 +91,68 @@ def __init__(self, *args, camera, camera_queue_mb=100, **kwargs):
         self.gui_timer.timeout.connect(self.send_gui_parameters)
         self.gui_timer.timeout.connect(self.acc_camera_framerate.update_list)
 
-    def reset(self):
+        self.recording = recording
+        if recording is not None:
+            self._setup_recording(
+                kbit_framerate=recording.get("kbit_rate", 1000),
+                extension=recording["extension"],
+            )
+
+    def reset(self) -> None:
         super().reset()
         self.acc_camera_framerate.reset()
 
-    def initialize_plots(self):
+    def initialize_plots(self) -> None:
         super().initialize_plots()
 
-    def send_gui_parameters(self):
+    def send_gui_parameters(self) -> None:
         self.camera.control_queue.put(self.camera_state.params.changed_values())
         self.camera_state.params.acknowledge_changes()
 
-    def start_experiment(self):
+    def start_experiment(self) -> None:
         """ """
         self.go_live()
         super().start_experiment()
 
-    def make_window(self):
+    def start_protocol(self) -> None:
+        """
+        Starts the recording if the recording parameters are set.
+        """
+        if self.recording is not None:
```

Comment:
so recording is either `None` or dict? Please type-annotate it as `Optional[Dict[str, Any]]` (the types you get from the Python `typing` standard module.

Reply:
Related to this, should be consider to slowly type-annotate the whole Stytra?